host.go

```diff
@@ -378,6 +387,60 @@ func (h *Host) InitCommands(c *chat.Commands) {
 		},
 	})
 
+	c.Add(chat.Command{
```

Comment:
Hmm I'd much prefer to keep the `/nick` command in the `chat` module.

Could we do the extra checking inside the `room.Rename` function?

Also preferably the Room shouldn't know about details of the Host's implementation, so ideally the Host would pass in something like a `CheckAvailableID(string) error` interface to the Room, and that can do the checking.

Reply:
The reason I moved /nick to host.go was so that the Command was able to get access to and Auth instance (through the host), which is where the reservation data lives. 

At first I saw that Host is an extension of a Room, and tried a type assertion on it, but that didn't seem to work.  Perhaps there's a way to do that without moving the nick Command. My Go knowledge isn't the best, so pointers welcome!